ansible_collections/arista/avd/plugins/module_utils/eos_designs.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from functools import cached_property
+try:
+    from ansible_collections.arista.avd.plugins.filter.convert_dicts import convert_dicts
+    from ansible_collections.arista.avd.plugins.filter.list_compress import list_compress
+    from ansible_collections.arista.avd.plugins.filter.range_expand import range_expand
+    from ansible_collections.arista.avd.plugins.filter.natural_sort import natural_sort
+except ImportError as imp_exc:
+    FILTER_IMPORT_ERROR = imp_exc
+else:
+    FILTER_IMPORT_ERROR = None
+
+
+class AVDFacts:
+
+    def __init__(
+        self,
+        hostvars=None,
+        combine=None,
+        template_lookup_module=None
+    ):
+        if hostvars:
+            self._hostvars = hostvars
+        if combine:
+            self._combine = combine
+        if template_lookup_module:
+            self._template_lookup_module = template_lookup_module
+        self._list_compress = list_compress
+        self._range_expand = range_expand
+        self._convert_dicts = convert_dicts
+        self._natural_sort = natural_sort
+
+    @classmethod
+    def keys(cls):
+        '''
+        Return the list of "keys"
+
+        Actually the returned list are the names of attributes not starting with "_" and using cached_property class.
+        The "_" check is added to allow support for "internal" cached_properties storing temporary values.
+        '''
+
+        return [key for key in dir(cls) if not key.startswith('_') and isinstance(getattr(cls, key), cached_property)]
+
+    @classmethod
+    def internal_keys(cls):
+        '''
+        Return the list of "keys" starting with _
+
+        Actually the returned list are the names of attributes starting with "_" and using cached_property class.
+        The "_" check is added to only include "internal" cached_properties storing temporary values.
+        '''
+
+        return [key for key in dir(cls) if key.startswith('_') and isinstance(getattr(cls, key), cached_property)]
+
+    def get(self, key, default_value=None):
+        '''
+        Emulate the builtin dict .get method
+        '''
+
+        if key in self.keys():
+            return getattr(self, key)
+        return default_value
+
+    def reset(self):
+        '''
+        Reset the cached attribute values
+        '''
+
+        self.__dict__ = {}
```

Comment:
Have we tested this? This would remove every single attribute not only the cached ones right?

I think we should rather do something like:

```
for key in self.keys():
  del self.__dict__[key]
for key in self.internal_keys():
  del self.__dict__[key]

```

Reply:
It will not remove attributes. It will only remove the underlying __dict__ construct, which is how @cached_property works. I have tested it once.

Reply:
As agreed with Claus in our chat @Shivani-chourasiya let's remove the reset method for now